### DISCUSSION TITLE
smtp, sending mails fails

### DISCUSSION THREAD

**Discussion by fedi-will** [mastodon/mastodon#22749] [Q&A] [answered]
Completely new mastodon installation on a single purpose vServer, installed according to officilal docs. 
Did not see errors during installation. 
Now I wanted to test sending mails, did not work, then I found this in the logs: (output a bit shortened)

```
journalctl -u mastodon-sidekiq | grep smtp
Dez 26 10:45:24 bundle[57391]: 2022-12-26T09:45:24.807Z pid=57391 tid=3ktxf \
  WARN: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `initialize'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `open'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `tcp_socket'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `initialize'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `open'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `tcp_socket'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:588:in `block in do_start'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:587:in `do_start'
bundle[57391]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:557:in `start'
bundle[57391]: /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/\ 
delivery_methods/smtp.rb:109:in `start_smtp_session'
bundle[57391]: /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/\ 
delivery_methods/smtp.rb:100:in `deliver!'
bundle[57391]: 2022-12-26T09:45:52.811Z pid=57391 tid=37ler \ 
  WARN: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:578:in `initialize'
[ same error repeated a few times ]
```

ufw is installed, but port 587out allowed.

I am out of ideas, any hints?

**Comment by rbairwell** [comment, marked as answer]
How are you sending the emails?

If you are sending them "locally" have you:
* Actually installed a SMTP server such as Postfix, Sendmail, Qmail, Exim on the server
* Configured it to listen on localhost to port 587 (if that is the configuration you have set in Mastodon)
* and allowed ports *25 AND 587* outbound for email delivery
* can you send a test email to yourself from the command line? What error do you get if it fails?

Or are you using a third party service such as Mailgun, Amazon SES, Google SMTP Relay etc etc?

How are you running Mastodon? Did you [install from source](https://docs.joinmastodon.org/admin/install/), using the DigitalOcean 1-Click install image, using [Vagrant](https://docs.joinmastodon.org/dev/setup/#vagrant), using the Docker/Docker-compose setups or other as detailed in the [repo](https://github.com/mastodon/mastodon)?

See:
* [Run your own server](https://docs.joinmastodon.org/user/run-your-own/#so-you-want-to-run-your-own-mastodon-server) 

> "An e-mail provider. Mastodon needs to send confirmation links and various notifications through e-mail, and hosting your own SMTP server, while possible, is much more difficult to do reliably than to use a third-party provider.
> How to get: Mailgun, SparkPost, Postmark, Sendgrid, any of the infinite number of e-mail hosting providers that expose a SMTP API. Comes with a monthly cost based on volume of e-mails sent.


* [Pre-requisites](https://docs.joinmastodon.org/admin/install/#pre-requisites) "An e-mail delivery service or other SMTP server"
* [Configuration](https://docs.joinmastodon.org/admin/config/#smtp)

**Reply by fedi-will** [reply]
Hello @rbairwell, 
thanks a lot for looking into my problem. 
As you suggested, my solution was to **allow port 25 outgoing**, along with 587.

For the record: I still only have "SMTP_PORT=587" in my .env file

So a too restrictive firewall was my problem, maybe this helps somebody

**Reply by A1vinSmith** [reply]
Hi @fedi-will , I used the Digital Ocean market place image. Only allow port 25 outgoing not working for me.
It only works when I allow all TCP ports.

**Reply by A1vinSmith** [reply]
Even though the port 25 failed on me. Port 53 save the day by lookup.
